config/jobs/kubernetes-csi/gen-jobs.sh

```diff
@@ -266,7 +266,7 @@ EOF
 
     for tests in non-alpha alpha; do
         for deployment in 1.14 1.15 1.16 1.17; do # must have a deploy/kubernetes-<version> dir in csi-driver-host-path
-            for kubernetes in 1.14.6 1.15.3 1.16.2; do # these versions must have pre-built kind images (see https://hub.docker.com/r/kindest/node/tags)
+            for kubernetes in 1.14.6 1.15.3 1.16.2 1.17.0; do # these versions must have pre-built kind images (see https://hub.docker.com/r/kindest/node/tags)
```

Comment:
let's also remove 1.14 deploy + kubernetes versions since it's now out of support.